SpatialGDK/Source/SpatialGDK/Private/Tests/SpatialView/ExpectedViewDelta.cpp

```diff
@@ -0,0 +1,194 @@
+// Copyright (c) Improbable Worlds Ltd, All Rights Reserved
+
+#pragma once
+
+#include "ComponentTestUtils.h"
+#include "SpatialView/ViewDelta.h"
+#include "Tests/SpatialView/ExpectedEntityDelta.h"
+#include "Tests/SpatialView/ExptectedViewDelta.h"
+
+using namespace SpatialGDK;
+
+ExpectedViewDelta& ExpectedViewDelta::AddEntityDelta(Worker_EntityId EntityId, EntityChangeType Status)
+{
+	EntityDeltas.Add(EntityId, { EntityId, Status == ADD, Status == REMOVE });
+	return *this;
+}
+
+ExpectedViewDelta& ExpectedViewDelta::AddComponentAdded(Worker_EntityId EntityId, ComponentData ComponentData)
+{
+	EntityDeltas[EntityId].ComponentsAdded.Push(ComponentChange(ComponentData.GetComponentId(), ComponentData.GetUnderlying()));
```

Comment:
This will actually point to the wrong thing at the end of the function. The ComponentData is passed by value, so ownership is transferred to this function, however you then don't move it into anything so it will go out of scope at the end of the function and delete the component data.

There are a few ways to deal with this:

1. Have ExpectedEntityDelta mirror the entity deltas and have several `TArray<ComponentChange>` objects, but also have a `TArray<ComponentData> DataStorage` and `TArray<ComponentUpdate> UpdateStorage` which hold onto these objects so that the ExpectedEntityDelta owns it's own data.
So you call GetUnderlying, and then `DataStorage.Add(MoveTemp(Data));`

1.  Merge the storage and the expectations together. So instead of having several `TArray<ComponentChange>` objects you have something like `TArray<ComponentData> ExpectedComponentAdds`.

1. Use Release instead of get underlying and manually clean things up in the destructor. This is my least favourite but added for completeness.